Carbon/components/FetchMonthEmissions.js

```diff
@@ -3,19 +3,33 @@ import { API_URL } from "../config/Api";
 const TIMEOUT_DURATION = 25000;
 
 /**
-  Fetches month emissions data from the API.
-  @param {string} month - The month for which to fetch emissions data.
-  @param {string} user_id - The ID of the user for whom to fetch emissions data.
-  @returns {Promise} A Promise that resolves with the fetched data, or rejects with an error.
-  @throws {Error} If there was an error fetching the data.
-  @example const data = await FetchMonthEmissions('january', '1234');
-**/
-export const FetchMonthEmissions = async(month, user_id) => {
+  Fetches the emissions data for a given user and year-month combination from the API.
+
+  @param {string} yearMonth - The year and month to fetch emissions data for in the format "YYYY-MM".
+  @param {number} user_id - The ID of the user to fetch emissions data for as a non-negative integer.
+  @returns {Promise} A promise that resolves with the fetched data or null if there is no data, or rejects with an error.
+  @throws {Error} If there is an issue with the inputs or there is an error fetching the data.
+
+  @example
+  const data = await FetchMonthEmissions('2023-01', 1234);
+*/
+export const FetchMonthEmissions = async(yearMonth, user_id) => {
   try {
+
+    // Check if yearMonth is in correct format
+    const isValidYearMonth = /^\d{4}-\d{2}$/.test(yearMonth);
+    if (!isValidYearMonth) {
+      throw new Error('FetchMonthEmissions: yearMonth is not in the correct format (YYYY-MM)');
+    }
+
+    if (!Number.isInteger(user_id) || user_id < 0) {
+      throw new Error('FetchMonthEmissions: user_id must be a non-negative whole number');
+    }
+
     // Construct the URL for the API endpoint
     // const url = `${API_URL}userEmissions/${month}/${user_id}`; // TODO: Connect to the actual API endpoint eventually
 
-    const url = `http://192.168.0.165:3000/api/userEmissions/${month}/${user_id}` // Using the local IPv4 for testing purposes
+    const url = `http://192.168.0.165:3000/api/userEmissions/${yearMonth}/${user_id}` // Using the local IPv4 for testing purposes
```

Comment:
Change this to API_URL and comment out the deployment url from API.js in order to clarify that this is a call thats not yet deployed.
Alternatively you could deploy the server with the changes that are on this branch, as they look good to me.

Reply:
Resolved in [28501cd](https://github.com/CivBuilder/Carbon/pull/88/commits/28501cdee91662a4ead773479a8770cb5159bffe)!